File: 02-Basic/JavaScript/05_binaryToDecimal.js
let string= "10111100001"

function binaryToDecimal(str){
    let decimal=0
    let power=0
    for (let i = str.length-1; i >=0; i--) {
        console.log(power)
      
        decimal += parseInt(str[i]) * (2 ** power);
        power++;
    }
    return decimal
// 
}

console.log(binaryToDecimal(string))

File: 02-Basic/python/05-binary_to_decimal.py
string= "1001100"

def binary_to_decimal(str):
    decimal=0
    power=0
    for i in range(len(str) - 1, -1, -1):
        print(f"Index: {i}, Power: {power}, Digit: {str[i]}")
        decimal += int(str[i]) * (2 ** power)
        power += 1
    return decimal
print(binary_to_decimal(string))